Can't publish new spring-boot packages referencing the being-published beeline version.

Attempting to update the `beelineVersion` referenced by the spring-boot-sleuth-starter and spring-boot-starter packages to the version of the beeline being release causes the release to [fail the publish to Maven](https://app.circleci.com/pipelines/github/honeycombio/beeline-java/966/workflows/5776ebeb-8562-47fb-922f-7ce07a4c3500/jobs/3324). In the first run at releasing 1.7.0:

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Beeline Java (Parent) 1.7.0:
[INFO] 
[INFO] Beeline Java (Parent) .............................. SUCCESS [  5.763 s]
[INFO] Beeline Java (Core) ................................ SUCCESS [  0.953 s]
[INFO] Beeline Java (Spring Boot Starter) ................. FAILURE [  1.697 s]
[INFO] Beeline Java (Spring Boot Sleuth Starter) .......... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.368 s
[INFO] Finished at: 2021-12-23T23:30:49Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project beeline-spring-boot-starter: Could not resolve dependencies for project
        io.*********.beeline:beeline-spring-boot-starter:jar:1.7.0: Could not find artifact
        io.*********.beeline:beeline-core:jar:1.7.0 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  maven: circleci/maven@1.0.3
+  maven: circleci/maven@1.3.0
 
 commands:
   with_cache:

